homeassistant/components/foscam/camera.py

```diff
@@ -108,6 +118,21 @@ def __init__(self, camera: FoscamCamera, config_entry: ConfigEntry) -> None:
         if self._rtsp_port:
             self._attr_supported_features = CameraEntityFeature.STREAM
 
+        self._name = config_entry.title
+
+        dev_info = (
+            coordinator.data["dev_info"] if "dev_info" in coordinator.data else None
+        )
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, dev_info["mac"] if dev_info else "_camera")},
+            manufacturer="Foscam",
+            model=dev_info["productName"] if dev_info else None,
+            name=config_entry.title,
+            sw_version=dev_info["firmwareVer"] if dev_info else None,
+            hw_version=dev_info["hardwareVer"] if dev_info else None,
+        )
```

Comment:
Not related to adding a coordinator. Please put that in a separate PR which should also implement `use_entity_name` and translations

Reply:
If this is not added, the coordinator is never used. So would like to keep this.

Reply:
Of course it is as it adds `available` etc. at properties + the fact that it holds the session used.
Adding `DeviceInfo` should not be in this PR